pkg/dns/aws/dns.go

```diff
@@ -157,6 +154,14 @@ func NewProvider(config Config, operatorReleaseVersion string) (*Provider, error
 		}
 	}
 
+	// When RoleARN is provided, make a copy of the Route53 session and configure it to use RoleARN.
+	// RoleARN is intended to only provide access to another account's Route53 service, not for ELBs.
+	sessRoute53 := sess
+	if config.RoleARN != "" {
+		sessRoute53 := sess.Copy(sess.Config)
```

Comment:
You shouldn't need to specify `sess.Config`.  https://github.com/openshift/cluster-ingress-operator/blob/b1a6bb5c8b9ee864fd54374bc833fc24b7b935fd/vendor/github.com/aws/aws-sdk-go/aws/session/session.go#L795-L797
```suggestion
		sessRoute53 := sess.Copy()
```
I'm not certain, but you might be able to hang this config on `r53Config` below instead:
```go
	r53Config := aws.NewConfig()
	// When RoleARN is provided, make a copy of the Route53 session and configure it to use RoleARN.
	// RoleARN is intended to only provide access to another account's Route53 service, not for ELBs.
	if config.RoleARN != "" {
		r53Config.WithCredentials(stscreds.NewCredentials(sess, config.RoleARN))
	}
```

Reply:
Done & will keep pattern in my back pocket.

Reply:
@Miciah Turns out, it appears that i need `sess.Config`, otherwise, it breaks.

Added `sessRoute53 := sess.Copy(sess.Config)` back, and in my smoke testing, it worked.

Reply:
Actually, I tested again, and it didn't work. I think I might of ran an invalid test, possibly the service hanging around while I created a new ingresscontroller caused me to get a false positive by somehow re-using the existing service...not sure, will update when I find what works.